addons/adyen_platforms/models/adyen_account.py

```diff
@@ -154,40 +175,88 @@ def _compute_full_name(self):
             else:
                 adyen_account_id.full_name = "%s %s" % (adyen_account_id.first_name, adyen_account_id.last_name)
 
+    @api.depends('kyc_tier', 'adyen_kyc_ids', 'account_status')
+    def _compute_state(self):
+        for account in self:
+            if account.account_status == 'inactive' and account.kyc_tier == 0:
+                account.state = 'pending'
+            else:
+                if account.adyen_kyc_ids.filtered(lambda k: k.status == 'awaiting_data'):
```

Comment:
Maybe use any() to stop filtering as soon as one record is found.

I suppose we can expect a lot of kyc's